frb_codegen/src/utils.rs

```diff
@@ -12,15 +12,69 @@ pub fn mod_from_rust_path(code_path: &str, crate_path: &str) -> String {
         .replace('/', "::")
 }
 
-pub fn with_changed_file<F: FnOnce() -> anyhow::Result<()>>(
+pub fn with_changed_file(
     path: &str,
     append_content: &str,
-    f: F,
+    generate_c_from_rust_func: impl FnOnce() -> anyhow::Result<()>,
+    generate_dart_from_c_func: impl FnOnce() -> anyhow::Result<()>,
 ) -> anyhow::Result<()> {
     let content_original = fs::read_to_string(&path)?;
     fs::write(&path, content_original.clone() + append_content)?;
 
-    f()?;
+    generate_c_from_rust_func()?;
+    generate_dart_from_c_func()?;
 
     Ok(fs::write(&path, content_original)?)
 }
+
+pub fn find_all_duplicates(mut vec: Vec<String>) -> anyhow::Result<()> {
+    let mut conflicts = vec![];
+    if vec.is_empty() {
+        return Ok(());
+    }
+    vec.sort();
+
+    for i in 0..(vec.len() - 1) {
+        if vec[i] == vec[i + 1] && conflicts.binary_search(&vec[i]).is_err() {
+            conflicts.push(vec[i].clone());
+        }
+    }
+
+    if !conflicts.is_empty() {
+        let mut conflict_symbols = conflicts
+            .iter()
+            .map(|s| s.clone() + ",")
+            .collect::<String>();
+        conflict_symbols.pop();
+
+        let (symbol_str, verb_str) = if conflicts.len() == 1 {
+            ("symbol", "has")
+        } else {
+            ("symbols", "have")
+        };
+        panic!(
+            "{} [{}] {} already been defined",
+            symbol_str, conflict_symbols, verb_str
+        );
+    }
+
+    Ok(())
+}
+
+/// check api defined by users, if no duplicates, then generate all symbols (api function name)
+/// ,including those generated implicitily by frb
```

Comment:
"," maybe at end of line, not begin of line

Reply:
fixed